sdk/storage/Azure.Storage.DataMovement/src/Models/TransferCheckpointerMethod.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Azure.Storage.DataMovement.Models
+{
+    /// <summary>
+    /// Base Checkpointer class to create the checkpointing logic
+    /// to resume from.
+    /// </summary>
+    public class TransferCheckpointerMethod
```

Comment:
The usage of this class seems a little odd to me, but I think I'm mostly just getting hung up on the naming. `TransferCheckpointerMethod` sounds like an `Enum` to me which I know we discussed we wanted a class instead. I think as is, a good name for this thing would be just `TransferCheckpointer` because to the customer, they are still sort "creating" a checkpointer with this class, and we just pick the right one to build. But `TransferCheckpointer` is taken :)

Reply:
Talked offline, will be changed to Options and will not be creating the transfer checkpointer / setting a transfer checkpointer in this class. Will be used to make the transfer checkpointer when constructing the transfermanager.

done.